File: src/client/src/app/shared/models/order.ts
export interface Order {
    id: number;
    buyerEmail: string;
    createdAt: string;
    shippingAddress: ShippingAddress;
    paymentSummary: PaymentSummary;
    deliveryMethod: string;
    orderItems: OrderItem[];
    subtotal: number;
    discount?: number;
    shippingPrice: number;
    total: number;
    invoice: string;
    status: string;
  }

  export interface ShippingAddress {
    name: string;
    line1: string;
    line2?: string;
    city: string;
    state: string;
    postalCode: string;
    country: string;
  }

  export interface PaymentSummary {
    last4: number;
    brand: string;
    expMonth: number;
    expYear: number;
  }
  
export interface OrderItem {
    productId: number;
    productName: string;
    imageUrl: string;
    price: number;
    quantity: number;
  }

export interface OrderToCreate {
    cartId: string;
    deliveryMethodId: number;
    shippingAddress: ShippingAddress;
    paymentSummary: PaymentSummary;
    discount?: number;
}

export class PaginationParams {
  pageNumber = 1;
  pageSize = 5;
  sort = '-createdAt';
} 

File: src/FlowerShop.DataAccess/Core/Entities/OrderAggregate/Order.cs
using FlowerShop.DataAccess.Core.Entities.Interfaces;
using FlowerShop.DataAccess.Core.Enums;

namespace FlowerShop.DataAccess.Core.Entities.OrderAggregate;

public class Order : IEntityBase
{
    public int Id { get; init; }
    public required string BuyerEmail { get; init; }
    public DateTime CreatedAt { get; init; }
    public ShippingAddress ShippingAddress { get; init; } = null!;
    public DeliveryMethod DeliveryMethod { get; set; } = null!;
    public PaymentSummary PaymentSummary { get; init; } = null!;
    public decimal Subtotal { get; init; }
    public decimal Discount { get; init; }
    public OrderState OrderState { get; init; }
    public string Invoice { get; set; } = string.Empty;
    public required string PaymentIntentId { get; init; }

    public List<OrderItem> OrderItems { get; init; } = [];
    public List<Reservation> Reservations { get; init; } = [];


    public decimal GetTotal()
    {
        return Subtotal - Discount + DeliveryMethod.Price;
    }
}